### PR TITLE
Bump up PHP version constraint.

### PR DESCRIPTION
**Pull request by ADmad**
|    Q          |   A
|-------------- | ------
| Documentation | yes/no
| Bugfix        | yes/no
| BC Break      | yes/no
| New Feature   | yes/no
| RFC           | yes/no
| QA            | yes/no

### CODE REVIEW

**Comment by ADmad** [comment]
Travis is AWOL.

**Comment by weierophinney** [comment]
> Travis is AWOL.

We're aware; we've used up our allotment of credit hours for the month. I've got a support request in; in the meantime, we'll be attempting to verify manually before merge.

**Comment by ADmad** [comment]
You'll have to switch to Github Actions (or another CI provider) eventually :slightly_smiling_face:.

**Comment by weierophinney** [comment]
> You'll have to switch to Github Actions (or another CI provider) eventually .

Already planned: https://github.com/laminas/technical-steering-committee/blob/main/meetings/minutes/2020-11-02-TSC-Minutes.md#github-actions

**Comment by ADmad** [comment]
If you want I can add a simple new workflow to run the testsuite on Github Action. You can later remove it or improve upon.

**Comment by weierophinney** [comment]
> If you want I can add a simple new workflow to run the testsuite on Github Action. You can later remove it or improve upon.

Would rather wait, as it will be easier to script adding the workflow(s) to repositories later if they don't already have ad-hoc changes.

**Comment by ADmad** [comment]
> Would rather wait, as it will be easier to script adding the workflow(s) to repositories later if they don't already have ad-hoc changes.

Sorry, I added one. After seeing the test suite passing on PHP 8, I removed it again :slightly_smiling_face:.

File: composer.json [hunk 1]
```diff
@@ -27,7 +27,7 @@
         }
     },
     "require": {
-        "php": "^7.1",
+        "php": "^7.2 || ~8.0.0",
```
Comment (weierophinney) [inline composer.json]: We can, and should, bump this to 7.3, as 7.2 goes out of support in 11 days.

File: composer.json [hunk 2]
```diff
@@ -36,7 +36,7 @@
     "require-dev": {
         "laminas/laminas-coding-standard": "~1.0.0",
         "laminas/laminas-diactoros": "^1.7 || ^2.1.1",
-        "phpunit/phpunit": "^7.0.2"
+        "phpunit/phpunit": "^8.5 || ^9.1"
```
Comment (weierophinney) [inline composer.json]: If we bump to PHP 7.3, we can bump this to just `^9.3`.

File: composer.json [hunk 3]
```diff
@@ -36,7 +36,7 @@
     "require-dev": {
         "laminas/laminas-coding-standard": "~1.0.0",
         "laminas/laminas-diactoros": "^1.7 || ^2.1.1",
```
Comment (weierophinney) [inline composer.json]: We can remove the 1.7 constraint here.

**Review by weierophinney — CHANGES_REQUESTED** [review]
I've flagged a few changes; I'll go ahead and make them and push them back after testing on each of the target versions locally.

File: composer.json (hunk 1 above)
Comment (boesing) [reply composer.json]: Actually, its PHP 7.3 which goes out of support in 11 days.
https://www.php.net/supported-versions

Comment (ADmad) [reply composer.json]: > Actually, its PHP 7.3 which goes out of support in 11 days.

True, but there's no need to drop support for PHP 7.3 just yet.

Comment (boesing) [reply composer.json]: > True, but there's no need to drop support for PHP 7.3 just yet.

Yup, just wanted to fix that message as it states that PHP 7.3 might be supported for another year which aint true.

**Comment by ADmad** [comment]
> I'll go ahead and make them and push them back after testing on each of the target versions locally.

Cool, I'll await your updates.

**Review by weierophinney — APPROVED** [review]

File: composer.json (hunk 1 above)
Comment (weierophinney) [reply composer.json]: @boesing Sorry, let me clarify: PHP 7.2 goes end-of-life in 11 days. PHP 7.3 continues to receive security patches for another year. Our security window is for versions that receive updates of any kind, so it's those getting both active support and those getting security patches. This is why I suggested bumping the minimum to 7.3.

**Comment by weierophinney** [comment]
I took the liberty of removing Prophecy mocks for native PHPUnit mocks... which required a fair bit of change to the `SapiStreamEmitterTest`, as it had a rather convoluted setup. I extracted much of it to a `MockStreamHelper` that provides state and callbacks that a mock stream instance can use, which leads to more explicit setup. All the original test cases continue to pass on both lowest and latest dependencies locally, so I'm going to go ahead and merge.

**Pull request merged**